test/ng/compileSpec.js

```diff
@@ -4033,6 +4033,68 @@ describe('$compile', function() {
     });
 
 
+    it('should bind to multiple directives controllers via object notation (new scope)', function() {
+      var controllerCalled = false;
+      var secondControllerCalled = false;
+      module(function($compileProvider, $controllerProvider) {
+        $controllerProvider.register('myCtrl', function() {
+          expect(this.data).toEqualData({
+            'foo': 'bar',
+            'baz': 'biz'
+          });
+          expect(this.str).toBe('Hello, world!');
+          expect(this.fn()).toBe('called!');
+          controllerCalled = true;
+        });
+        $controllerProvider.register('secondCtrl', function() {
+          expect(this.data).toEqualData({
+            'foo2': 'bar2',
+            'baz2': 'biz2'
+          });
+          expect(this.str).toBe('Hello, second world!');
+          expect(this.fn()).toBe('second called!');
+          secondControllerCalled = true;
+        });
+        $compileProvider.directive('fooDir', valueFn({
+          bindToController: {
+            'data': '=dirData',
+            'str': '@dirStr',
+            'fn': '&dirFn'
+          },
+          scope: true,
+          controller: 'myCtrl as myCtrl'
+        }));
+        $compileProvider.directive('barDir', valueFn({
+          bindToController: {
+            'data': '=barData',
+            'str': '@barStr',
+            'fn': '&barFn'
+          },
+          scope: true,
```

Comment:
We do support multiple directives asking for new normal scope (i.e. not isolate) as can be seen in the docs (https://docs.angularjs.org/api/ng/service/$compile#-scope-):

> The scope property can be true, an object or a falsy value:
>
>**true**: A new child scope that prototypically inherits from its parent will be created for the directive's element. If multiple directives on the same element request a new scope, only one new scope is created. The new scope rule does not apply for the root of the template since the root of the template always gets a new scope.

So this is a valid bug and the fix looks pretty good to me too. I will give it a review and land this week.